File: leetcode/100-199/102_binary_tree_level_order_traversal.py

def levelOrder(self, root: TreeNode) -> List[List[int]]:
    if not root:
        return []
    
    result = []
    
    q = collections.deque()
    q.append(root)
    
    while q:
        level = []
        level_length = len(q)
        for i in range(level_length):
            node = q.popleft()
            level.append(node.val)
            
            if node.left:
                q.append(node.left)
                
            if node.right:
                q.append(node.right)
                
        result.append(level)
        
    return result

File: leetcode/100-199/102_binary_tree_level_order_traversal.cpp
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        
        vector<vector<int>> result;
        
        if (root ==NULL) {
            return result;
        }
        
        queue<TreeNode*> q;
        
        q.push(root);
        
        while(!q.empty()) {
            
            vector<int> level;
            int level_length = q.size();
            
            for(int i = 0; i < level_length; i++) {
                TreeNode* node = q.front();
                q.pop();
                level.push_back(node->val);
                if(node->left != NULL) {
                    q.push(node->left);
                    
                }
                if(node->right != NULL) {
                    q.push(node->right);
                }
            }
            result.push_back(level);   
        }
        return result;
    }
};